tests/integration2/pilot/sidecar_api_test.go

```diff
@@ -1,40 +1,34 @@
 package pilot
 
 import (
-	"fmt"
-	"net"
+	"path/filepath"
 	"testing"
 	"time"
 
+	"istio.io/istio/pkg/test/framework2"
+	"istio.io/istio/pkg/test/framework2/components/galley"
+	pilot2 "istio.io/istio/pkg/test/framework2/components/pilot"
+
 	xdsapi "github.com/envoyproxy/go-control-plane/envoy/api/v2"
 	xdscore "github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
 
 	"istio.io/istio/pilot/pkg/model"
 	"istio.io/istio/pkg/log"
-	"istio.io/istio/pkg/test/framework"
-	"istio.io/istio/pkg/test/framework/api/components"
-	"istio.io/istio/pkg/test/framework/api/descriptors"
-	"istio.io/istio/pkg/test/framework/api/ids"
-	"istio.io/istio/pkg/test/framework/api/lifecycle"
 	"istio.io/istio/pkg/test/scopes"
 	"istio.io/istio/pkg/test/util/structpath"
 )
 
 func TestSidecarListeners(t *testing.T) {
 	// Call Requires to explicitly initialize dependencies that the test needs.
-	ctx := framework.GetContext(t)
+	ctx := framework2.NewContext(t)
```

Comment:
With the new context API, you'll need to call defer ctx.Done()

Reply:
done

Reply:
err.. not here?